### PR TITLE
test(support): verify support tests results in all tested browsers

### PR DESCRIPTION
**Pull request by mgol**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

New tests.

**What is the current behavior? (You can also link to an open issue here)**

It's not tested whether browsers pass or fail specific support tests used to skip particular test blocks.

**What is the new behavior (if this is a feature change)?**

All browsers have verified support tests results.

**Does this PR introduce a breaking change?**

No.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- ~~Docs have been added / updated (for bug fixes / features)~~

**Other information**:

### CODE REVIEW

File: test/ng/supportSpec.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+'use strict';
```
Comment (gkalpak) [inline test/ng/supportSpec.js]: I would expect this file to be in `test/helpers/`. Why here? It is confusing.

File: test/ng/supportSpec.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+describe('support test results', function() {
+  var expected, version, testName;
+  var userAgent = window.navigator.userAgent;
+
+  // Support: iOS 8 only
+  if (/iPhone OS 10_1\d(?:_\d+)? /.test(userAgent)) {
+    // iOS 8 official simulators have broken user agent (containing something like `iPhone OS 10_12_5`,
+    // i.e. the macOS version in place of the iOS version so they'd fall into an incorrect bucket.
```
Comment (gkalpak) [inline test/ng/supportSpec.js]: Missing closing `)`.

File: test/ng/supportSpec.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    // Fix the user agent there.
+    // NOTE: Make sure the above check doesn't catch the real iOS 10!
+    userAgent = userAgent.replace(/iPhone OS 10(?:_\d)+/, 'iPhone OS 8_1');
```
Comment (gkalpak) [inline test/ng/supportSpec.js]: `_\d` --> `_\d+`

File: test/ng/supportSpec.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  }
+
+  if (/edge\//i.test(userAgent)) {
```
Comment (gkalpak) [inline test/ng/supportSpec.js]: Why not convert `userAgent` to lowercase once and avoid all the `i` flags? :smiley:

File: test/ng/supportSpec.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    expected = {
+      classes: true,
+      fatArrows: true,
+      shorthandMethods: true
+    };
+  } else if (/(?:msie|trident)/i.test(userAgent)) {
```
Comment (gkalpak) [inline test/ng/supportSpec.js]: `/(?:msie|trident)/` --> `/msie|trident/`

File: test/ng/supportSpec.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    };
+  } else if (/chrome/i.test(userAgent)) {
+    // Catches Chrome on Android as well (i.e. the default
+    // Android browser on Android >= 4.4).
+    expected = {
+      classes: true,
+      fatArrows: true,
+      shorthandMethods: true
+    };
+  } else if (/\b9\.\d+(\.\d+)* safari/i.test(userAgent)) {
```
Comment (gkalpak) [inline test/ng/supportSpec.js]: `/\b9\.\d+(\.\d+)*` --> `/\b9(?:\.\d+)+`

File: test/ng/supportSpec.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    expected = {
+      classes: true,
+      fatArrows: false,
+      shorthandMethods: true
+    };
+  } else if (/\b\d+\.\d+(\.\d+)* safari/i.test(userAgent)) {
```
Comment (gkalpak) [inline test/ng/supportSpec.js]: `/\b\d+\.\d+(\.\d+)*` --> `/\b\d+(?:\.\d+)+`

File: test/ng/supportSpec.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+      fatArrows: false,
+      shorthandMethods: true
+    };
+  } else if (/iphone os/i.test(userAgent)) {
+    expected = {
+      classes: true,
+      fatArrows: true,
+      shorthandMethods: true
+    };
+  } else if (/android 4\.[0-3]/i.test(userAgent)) {
```
Comment (gkalpak) [inline test/ng/supportSpec.js]: This will break when Android 4.10 is out :stuck_out_tongue:

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: test/ng/supportSpec.js (hunk 1 above)
Comment (mgol) [inline test/ng/supportSpec.js]: That was my plan. I must have clicked the wrong directory. ;)

File: test/ng/supportSpec.js (hunk 8 above)
Comment (mgol) [inline test/ng/supportSpec.js]: Android 4.x is dead, we're safe. :)

File: test/ng/supportSpec.js (hunk 1 above)
Comment (mgol) [inline test/ng/supportSpec.js]: Why in `test/helpers`? `test/helpers/` contains helpers for tests, not actual tests.

File: test/ng/supportSpec.js (hunk 4 above)
Comment (mgol) [inline test/ng/supportSpec.js]: That would have to be `lowerCasedUserAgent` to be precise. And maybe we'll need case sensitivity one day?

File: test/ng/supportSpec.js (hunk 1 above)
Comment (gkalpak) [inline test/ng/supportSpec.js]: It contains helpers for tests and their tests (when they exist) :smiley:
For lack of a better directory, keeping the pec file next to the source is better than putting it into an arbitrary directory 😛

**Comment by mgol** [comment]
@gkalpak PR updated, tests now pass.

**Comment by gkalpak** [comment]
Why did tests previously fail?
LGTM except for `supportSpec` being in `test/ng/` (instead of `test/helpers/`), as is [privateMocksSpec.js][1] for example.

[1]: https://github.com/mgol/angular.js/blob/492cb3537522da51ed6ea0f8895fc929ff8add17/test/helpers/privateMocksSpec.js

**Comment by mgol** [comment]
The spec file moved (I missed `privateMocksSpec.js` previously).

The tests were failing as I didn't add results for Safari 8 so it was treated the same as Safari 10. :-)

**Review by gkalpak — APPROVED** [review]

**Pull request merged**